src/core/thread/mesh_forwarder.cpp

```diff
@@ -1045,6 +1045,18 @@ void MeshForwarder::HandleReceivedFrame(Mac::RxFrame &aFrame)
             LogFrame("Received empty payload frame", aFrame, OT_ERROR_NONE);
         }
 
+#if OPENTHREAD_THREAD_VERSION >= OPENTHREAD_THREAD_VERSION_1_2 && OPENTHREAD_FTD
```

Comment:
Can we consider moving this whole check in `Mac::HandleReceivedFrame()` right before we call `Get<MeshForwarder>().HandleReceivedFrame(*aFrame);`.
The advantage I see with moving this is that we already have `neighbor` in the `Mac` (so no need to do `FindChild`).

Thoughts?

Reply:
Brilliant!